### PR TITLE
CAD-3444 mux changes from p2p-master

### PR DESCRIPTION
**Pull request by coot**
- network-mux: label mini-protocol thread
- network-mux: improve attenuated channel errors
- network-mux: SDUSize show via Quiet
- network-mux: mux runtime exceptions

### CODE REVIEW

**Review by karknu — APPROVED** [review]
LGTM

**Comment by coot** [comment]
bors merge

File: network-mux/src/Network/Mux/Types.hs [hunk 1]
```diff
@@ -255,3 +258,15 @@ muxBearerAsChannel bearer ptclNum ptclDir =
...
       noTimeout :: TimeoutFn m
       noTimeout _ r = Just <$> r
+
+--
+-- Errors
+--
+
+data MuxRuntimeError =
+    ProtocolAlreadyRunning    !MiniProtocolNum !MiniProtocolDir !MiniProtocolStatus
+  | UnknownProtocol           !MiniProtocolNum !MiniProtocolDir
```
Comment (coot) [inline network-mux/src/Network/Mux/Types.hs:268]: This might be caused by a frame with unknown protocol, in which case it shouldn't be here.

Comment (coot) [inline network-mux/src/Network/Mux/Types.hs:268]: `UnknownProtocolInternalError`

**Pull request merged**